test/cadet_web/controllers/user_controller_test.exs

```diff
@@ -12,31 +14,122 @@ defmodule CadetWeb.UserControllerTest do
 
   describe "GET /user" do
     @tag authenticate: :student
-    test "success, student", %{conn: conn} do
+    test "success, student non-story fields", %{conn: conn} do
       user = conn.assigns.current_user
       assessment = insert(:assessment, %{is_published: true})
       question = insert(:question, %{assessment: assessment})
       submission = insert(:submission, %{assessment: assessment, student: user})
       insert(:answer, %{question: question, submission: submission, grade: 50, adjustment: -10})
 
-      conn = get(conn, "/v1/user", nil)
-      body = json_response(conn, 200)
-      assert response(conn, 200)
-      assert %{"name" => user.name, "role" => "#{user.role}", "grade" => 40} == body
+      resp =
+        conn
+        |> get("/v1/user")
+        |> json_response(200)
+        |> Map.delete("story")
+
+      expected = %{"name" => user.name, "role" => "#{user.role}", "grade" => 40}
+
+      assert expected == resp
+    end
+
+    @tag authenticate: :student
+    test "success, student story ordering", %{conn: conn} do
+      early_assessments = build_assessments_starting_at(Timex.now())
+      late_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: 1))
+
+      for assessment <- early_assessments ++ late_assessments do
+        story =
+          conn
+          |> get("/v1/user")
+          |> json_response(200)
+          |> Map.get("story")
+
+        assert story == assessment.story
+
+        {:ok, _} = Repo.delete(assessment)
+      end
+    end
+
+    @tag authenticate: :student
+    test "success, student story skips assessment without story", %{conn: conn} do
+      assessments = build_assessments_starting_at(Timex.now())
+
+      assessments
+      |> Enum.fetch!(0)
+      |> Assessment.changeset(%{story: nil})
+      |> Repo.update()
+
+      story =
+        conn
+        |> get("/v1/user")
+        |> json_response(200)
+        |> Map.get("story")
+
+      assert story == Enum.fetch!(assessments, 1).story
+    end
+
+    @tag authenticate: :student
+    test "success, student story skips submitted", %{conn: conn} do
+      user = conn.assigns.current_user
+
+      early_assessments = build_assessments_starting_at(Timex.now())
+      late_assessments = build_assessments_starting_at(Timex.shift(Timex.now(), days: 1))
+      assessments_with_indices = Enum.with_index(early_assessments ++ late_assessments)
+
+      # Submit for i-th assessment, expect (i+1)th story to be returned
+      for {assessment, idx} <-
+            Enum.slice(assessments_with_indices, 0, length(assessments_with_indices) - 1) do
```

Comment:
I think `Enum.drop(assessments_with_indices, -1)` conveys the intent clearer

Reply:
Done